kernel/file.c

```diff
@@ -30,17 +29,15 @@ struct file*
 filealloc(void)
 {
   struct file *f;
-
-  acquire(&ftable.lock);
-  for(f = ftable.file; f < ftable.file + NFILE; f++){
-    if(f->ref == 0){
-      f->ref = 1;
-      release(&ftable.lock);
-      return f;
-    }
+  void* alloced_memory = bd_malloc(sizeof(struct file));
+  // Cant allocate sizeof(struct file) bytes
+  if (alloced_memory == 0){
+    return 0;
   }
-  release(&ftable.lock);
-  return 0;
+  f = (struct file*) alloced_memory;
```

Comment:
Fill the file with zeroes to preserve the same behavior as was before

Reply:
Valid comment really not sure why it has not affected programm on any test